pkg/ring/util.go

```diff
@@ -2,23 +2,26 @@ package ring
 
 import (
 	"math/rand"
-	"sort"
 	"time"
 )
 
-// GenerateTokens make numTokens random tokens, none of which clash
-// with takenTokens.  Assumes takenTokens is sorted.
+// GenerateTokens make numTokens unique random tokens, none of which clash
+// with takenTokens.
 func GenerateTokens(numTokens int, takenTokens []uint32) []uint32 {
 	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	added := make(map[uint32]bool)
```

Comment:
Nit: I would a prefer a name like `taken` or `existing`; to me `added` means that we have added them in this round.

Reply:
What about 'used' then? It's shortest I can come up with.

Reply:
`used` sgtm